crates/iota-sdk-crypto/src/lib.rs

```diff
@@ -197,32 +197,29 @@ pub const DERIVATION_PATH_PURPOSE_SECP256K1: u32 = 54;
 #[cfg(feature = "mnemonic")]
 pub const DERIVATION_PATH_PURPOSE_SECP256R1: u32 = 74;
 
-/// Defines a type which can be converted to bytes
-pub trait ToBytes {
-    /// Returns the raw bytes of this type.
-    fn to_bytes(&self) -> Vec<u8>;
+/// Defines the scheme of a private key
+pub trait PrivateKeyScheme {
+    const SCHEME: iota_types::SignatureScheme;
+
+    /// Returns the signature scheme for this private key
+    fn scheme(&self) -> iota_types::SignatureScheme {
+        Self::SCHEME
+    }
 }
 
 /// Defines a type which can be constructed from bytes
-pub trait FromBytes {
+pub trait ToFromBytes {
     type Error;
 
+    /// Returns the raw bytes of this type.
+    fn to_bytes(&self) -> Vec<u8>;
```

Comment:
is returning a slice not possible? 

Since it's now `to` and `from` maybe the docs could be something like:
```
/// Defines a type which can be constructed from and turned into bytes
```

Reply:
No, because we cannot use generics in the trait for the length of the array. But we could maybe have `as_bytes` which returns a `&[u8]`, I'll take a look